docs/DATABASE_FORMAT.md

```diff
@@ -0,0 +1,114 @@
+  
+  
+## NODE MANAGEMENT AND REDUNDANT EDGES  
+It is important to note that a database does not need to have every connection betwen two nodes defined in order to work. If, for example, we have three nodes A, B, and C, where the player can travel from any one node to any other, it is not necessary to explicitly permit the player to travel from A to C. If they can travel from A to B and then from B to C, the generator understands they can also travel from A to C. In fact, you should define the minimum feasible number of node connections when writing logic. This makes logic easier to read and edit, as well as being more optimal for generation.  
+  
+Sometimes, rooms exist in which every node connects to every other node in some unique and complicated way, such that no simplifications exist. In this case, often the logic involved can be made easier to work with by adding a node to an arbitrary location in the room. By adding extra nodes, you can represent in-between points between the existing nodes in a room, which is often helpful to reduce the number of possible scenarios that must be considered and to make the room's logic more organized.  
+  
+  
+![Redundant Edges](./img/redundant_mesh.png)  
+  
+  
+## NAMING CONVENTIONS  
+Area names should be memorable and descriptive. If your game does not provide an official name for a room, you will need to name it yourself. Avoid using in-jokes that newer players may be unfamiliar with, and remember that while a funny room name now and then can be good, it should not come at the expense of utility. Needless to say, profane or offensive names will be rejected outright. If a game's community has already adopted community names for rooms, it is best to use those as well, as many resources will have already been created using them, and players familiar to the game will easily adapt to your database. It can occasionally be worthwhile to alter or ignore an established community name, however, if it comes into conflict with the aforementioned principles.  
+  
+Node names do not need to be memorable, and should be purely descriptive. Because an area contains only a few nodes, anyone reviewing the database will almost certainly know what each node refers to. In addition, there are a few naming conventions that exist for specific node types:  
+  
+Pickup nodes should follow the format  
+Pickup (Pickup Type)  
+  
+If two pickups in a room would share the same name, it is required that you add some descriptive text to their names to distinguish them. For example,  
+Pickup (Lower Ledge Missile Expansion)  
+Pickup (Morph Tunnel Missile Expansion)  
+  
+Event nodes should follow the format  
+Event - Event Name  
+  
+And doors should adhere to  
+Door to Room Name  
```

Comment:
Docks are a bit more complex (depends on the dock type, uses "from" if the req is impossible).

Mentioning the namer in the editor is good, as well how if there's multiple pairs of dock between two areas (double path in Echoes) you need to add a suffix